### PR TITLE
Python 3: port autotest.client.utils.AsyncJob()

### PR DESCRIPTION
**Pull request by czhang0**
Signed-off-by: cuzhang <cuzhang@redhat.com>

### CODE REVIEW

File: virttest/utils_misc.py [hunk 1]
```diff
@@ -25,6 +25,12 @@
 import platform
 import traceback
 import math
+import select
+
+try:
+    from io import BytesIO
+except:
```
Comment (hsmj1412) [inline virttest/utils_misc.py]: except ImportError:

Comment (czhang0) [reply virttest/utils_misc.py]: I just saw your comment, yes, I modified same with you comments.
Thanks.

File: virttest/utils_misc.py [hunk 2]
```diff
@@ -4085,3 +4093,272 @@ def get_model_features(model_name):
...
+                 stdin=None, stderr_level=DEFAULT_STDERR_LEVEL,
+                 close_fds=False):
+        self.command = command
+        self.stdout_tee = get_stream_tee_file(stdout_tee, DEFAULT_STDOUT_LEVEL,
+                                              prefix=STDOUT_PREFIX)
+        self.stderr_tee = get_stream_tee_file(stderr_tee, stderr_level,
+                                              prefix=STDERR_PREFIX)
+        self.result = CmdResult(command)
+
+        # allow for easy stdin input by string, we'll let subprocess create
```
Comment (chunfuwen) [inline virttest/utils_misc.py]: allow/Allow

File: virttest/utils_misc.py [hunk 3]
```diff
@@ -4085,3 +4093,272 @@ def get_model_features(model_name):
...
+class AsyncJob(BgJob):
+
+    def __init__(self, command, stdout_tee=None, stderr_tee=None, verbose=True,
+                 stdin=None, stderr_level=DEFAULT_STDERR_LEVEL, kill_func=None,
+                 close_fds=False):
+        super(AsyncJob, self).__init__(command, stdout_tee=stdout_tee,
+                                       stderr_tee=stderr_tee, verbose=verbose, stdin=stdin,
+                                       stderr_level=stderr_level, close_fds=close_fds)
+
+        # start time for CmdResult
```
Comment (chunfuwen) [inline virttest/utils_misc.py]: start/Start

File: virttest/utils_misc.py [hunk 4]
```diff
@@ -4085,3 +4093,272 @@ def get_model_features(model_name):
...
+
+        if kill_func is None:
+            self.kill_func = self._kill_self_process
+        else:
+            self.kill_func = kill_func
+
+        if self.string_stdin:
+            self.stdin_lock = threading.Lock()
+            string_stdin = self.string_stdin
+            # replace with None so that _wait_for_commands will not try to re-write it
```
Comment (chunfuwen) [inline virttest/utils_misc.py]: replace/Replace

Comment (czhang0) [reply virttest/utils_misc.py]: Adopt.

File: virttest/utils_misc.py (hunk 3 above)
Comment (czhang0) [reply virttest/utils_misc.py]: Adopt and modify other same spell issue.

File: virttest/utils_misc.py (hunk 2 above)
Comment (czhang0) [reply virttest/utils_misc.py]: Adopt and modify other same spell issue.

File: virttest/utils_misc.py [hunk 5]
```diff
@@ -41,12 +47,14 @@
...
 from avocado.utils import memory
 from avocado.utils.astring import string_safe_encode
+from avocado.utils.process import CmdResult
 
 from . import data_dir
 from . import error_context
 from . import cartesian_config
 from . import utils_selinux
 from . import utils_disk
+from . import logging_manager
```
Comment (kylazhang) [inline virttest/utils_misc.py]: I think we need to change all relative path to absolute path? @chunfuwen

Comment (chunfuwen) [reply virttest/utils_misc.py]: @czhang0 ,please take this opportunities to replace all relative imports in this file

Comment (czhang0) [reply virttest/utils_misc.py]: Ok, I will use absolute import here.

**Pull request merged**